### ISSUE TITLE
[BUG] Compilation fails with gcc version 9.3.0

### ISSUE THREAD

**Issue by hyperbolic2346** [rapidsai/cudf#6085]
**Describe the bug**
By default on Ubuntu 20.04 gcc version 9.3.0 is installed. This version of gcc complains about narrowing in test files for orc and parquet.

```c++
  718 |   T c2b[] = {9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9,
      |     ^~~
/home/mwilson/code/upstream/cpp/tests/io/orc_test.cpp:718:5: error: narrowing conversion of ‘9’ from ‘int’ to ‘T’ {aka ‘bool’} [-Wnarrowing]
/home/knobby/code/upstream/cpp/tests/io/parquet_test.cpp:886:5: error: narrowing conversion of ‘8’ from ‘int’ to ‘T’ {aka ‘bool’} [-Wnarrowing]
```
This is not an issue if compiled with gcc 8.4.0, which can be installed with `apt install gcc-8` and then the gcc version can be swapped with update-alternatives. Set up with:
```
sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 9
sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 9
sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-8 8
sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-8 8
```
The active compiler can then be switched with `sudo update-alternatives --config gcc`.

**Steps/Code to reproduce bug**
Install gcc 9 and compile cudf. Alternatively a VM based on focal should provide the required gcc.

**Expected behavior**
Compilation to succeed.

**Environment overview (please complete the following information)**
bare-metal install with conda.

**Environment details**
```
 ../../print_env.sh 
<details><summary>Click here to see environment details</summary><pre>
     
     **git***
     commit 4b6b7c0fdb14aec638173201d04078a060c0eab4 (HEAD -> branch-0.15, upstream/branch-0.15, origin/branch-0.15, origin/HEAD)
     Merge: 1a3b3f217 ae650ad0b
     Author: Mike Wendt <mike.wendt@me.com>
     Date:   Mon Aug 24 14:17:30 2020 -0400
     
     Merge pull request #6058 from dillon-cullinan/fix-removemaster
     
     [REVIEW] Update docs to remove references to master [skip ci]
     **git submodules***
     
     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=20.04
     DISTRIB_CODENAME=focal
     DISTRIB_DESCRIPTION="Ubuntu 20.04.1 LTS"
     NAME="Ubuntu"
     VERSION="20.04.1 LTS (Focal Fossa)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 20.04.1 LTS"
     VERSION_ID="20.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=focal
     UBUNTU_CODENAME=focal
     Linux thermal-load 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Tue Aug 25 04:11:18 2020
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 440.95.01    Driver Version: 440.95.01    CUDA Version: 10.2     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |===============================+======================+======================|
     |   0  Quadro RTX 6000     Off  | 00000000:65:00.0 Off |                  Off |
     | 33%   33C    P8     3W / 260W |     18MiB / 24217MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     
     +-----------------------------------------------------------------------------+
     | Processes:                                                       GPU Memory |
     |  GPU       PID   Type   Process name                             Usage      |
     |=============================================================================|
     |    0      1249      G   /usr/lib/xorg/Xorg                             9MiB |
     |    0      1598      G   /usr/bin/gnome-shell                           7MiB |
     +-----------------------------------------------------------------------------+
     
     ***CPU***
     Architecture:                    x86_64
     CPU op-mode(s):                  32-bit, 64-bit
     Byte Order:                      Little Endian
     Address sizes:                   46 bits physical, 48 bits virtual
     CPU(s):                          12
     On-line CPU(s) list:             0-11
     Thread(s) per core:              2
     Core(s) per socket:              6
     Socket(s):                       1
     NUMA node(s):                    1
     Vendor ID:                       GenuineIntel
     CPU family:                      6
     Model:                           85
     Model name:                      Intel(R) Core(TM) i7-7800X CPU @ 3.50GHz
     Stepping:                        4
     CPU MHz:                         1200.189
     CPU max MHz:                     4000.0000
     CPU min MHz:                     1200.0000
     BogoMIPS:                        6999.82
     Virtualization:                  VT-x
     L1d cache:                       192 KiB
     L1i cache:                       192 KiB
     L2 cache:                        6 MiB
     L3 cache:                        8.3 MiB
     NUMA node0 CPU(s):               0-11
     Vulnerability Itlb multihit:     KVM: Mitigation: Split huge pages
     Vulnerability L1tf:              Mitigation; PTE Inversion; VMX conditional cache flushes, SMT vulnerable
     Vulnerability Mds:               Mitigation; Clear CPU buffers; SMT vulnerable
     Vulnerability Meltdown:          Mitigation; PTI
     Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
     Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
     Vulnerability Spectre v2:        Mitigation; Full generic retpoline, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling
     Vulnerability Srbds:             Not affected
     Vulnerability Tsx async abort:   Mitigation; Clear CPU buffers; SMT vulnerable
     Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti ssbd mba ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req md_clear flush_l1d
     
     ***CMake***
     /snap/bin/cmake
     cmake version 3.18.2
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 8.4.0-3ubuntu2) 8.4.0
     Copyright (C) 2018 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2019 NVIDIA Corporation
     Built on Sun_Jul_28_19:07:16_PDT_2019
     Cuda compilation tools, release 10.1, V10.1.243
     
     ***Python***
     /home/mwilson/miniconda3/bin/python
     Python 3.8.3
     
     ***Environment Variables***
     PATH                            : /home/mwilson/miniconda3/bin:/home/mwilson/miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
     LD_LIBRARY_PATH                 :
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/mwilson/miniconda3
     PYTHON_PATH                     :
     
     ***conda packages***
     /home/mwilson/miniconda3/bin/conda
     # packages in environment at /home/mwilson/miniconda3:
     #
     # Name                    Version                   Build  Channel
     _libgcc_mutex             0.1                        main
     ca-certificates           2020.1.1                      0
     certifi                   2020.4.5.1               py38_0
     cffi                      1.14.0           py38he30daa8_1
     chardet                   3.0.4                 py38_1003
     conda                     4.8.3                    py38_0
     conda-package-handling    1.6.1            py38h7b6447c_0
     cryptography              2.9.2            py38h1ba5d50_0
     idna                      2.9                        py_1
     ld_impl_linux-64          2.33.1               h53a641e_7
     libedit                   3.1.20181209         hc058e9b_0
     libffi                    3.3                  he6710b0_1
     libgcc-ng                 9.1.0                hdf63c60_0
     libstdcxx-ng              9.1.0                hdf63c60_0
     ncurses                   6.2                  he6710b0_1
     openssl                   1.1.1g               h7b6447c_0
     pip                       20.0.2                   py38_3
     pycosat                   0.6.3            py38h7b6447c_1
     pycparser                 2.20                       py_0
     pyopenssl                 19.1.0                   py38_0
     pysocks                   1.7.1                    py38_0
     python                    3.8.3                hcff3b4d_0
     readline                  8.0                  h7b6447c_0
     requests                  2.23.0                   py38_0
     ruamel_yaml               0.15.87          py38h7b6447c_0
     setuptools                46.4.0                   py38_0
     six                       1.14.0                   py38_0
     sqlite                    3.31.1               h62c20be_1
     tk                        8.6.8                hbc83047_0
     tqdm                      4.46.0                     py_0
     urllib3                   1.25.8                   py38_0
     wheel                     0.34.2                   py38_0
     xz                        5.2.5                h7b6447c_0
     yaml                      0.1.7                had09818_2
     zlib                      1.2.11               h7b6447c_3
     
</pre></details>
```

**Additional context**

**Comment by harrism** [comment]
@hyperbolic2346 @rongou is this fixed by (already merged) #6124 ?

**Comment by rongou** [comment]
Yes.

**Comment by harrism** [comment]
Fixed by #6124
